enterprise/server/auth/auth.go

```diff
@@ -313,33 +319,19 @@ func (c *apiKeyGroupCache) Add(apiKey string, apiKeyGroup interfaces.APIKeyGroup
 	c.mu.Unlock()
 }
 
+type Options struct {
+	AuthenticatorsOverride []authenticator // FOR TESTING
```

Comment:
Another option, instead of introducing test-only fields in constructors etc, is to put your test in the same namespace as the thing under test, then you can modify it directly. I'm not saying that's necessarily better here -- I usually like to test only the public interface anyway, but just putting it out there :) 

Even another option might be to introduce a NewOpenIDAuthenticatorForTesting method that just injects your test authenticators. It's clearly test-only, and minimally invasive (doesn't require changing all NewOpenIDAuthenticator callsites)

Reply:
Done. Introduced a test-only constructor.